components/EditCarForm/index.js

```diff
@@ -64,27 +63,31 @@ export default function EditCarForm({ onSubmit, form, activeCar }) {
             {Object.keys(carPrototype).map((attribute, index) => {
               if (group.content.includes(attribute)) {
                 const attributeValue = carPrototype[attribute];
-
                 const type =
                   typeof attributeValue === "string" ? "text" : "number";
                 return (
                   <div key={group.id + index}>
                     <label>
                       {attribute}:
-                      <input
-                        name={attribute}
-                        type={type}
-                        maxLength={
-                          attribute === "ImageUrl"
-                            ? null
-                            : type === "text"
-                            ? "17"
-                            : null
-                        }
-                        defaultValue={
-                          form === "edit" ? activeCar[attribute] : null
-                        }
-                      ></input>
+                      {type === "text" ? (
+                        <input
+                          name={attribute}
+                          type="text"
+                          maxLength="17"
+                          defaultValue={
+                            form === "edit" ? activeCar[attribute] : null
+                          }
+                        ></input>
+                      ) : (
+                        <input
+                          name={attribute}
+                          type="number"
+                          max={2000000}
+                          defaultValue={
+                            form === "edit" ? activeCar[attribute] : null
+                          }
+                        ></input>
+                      )}
```

Comment:
Your EditCarForm has a somewhat not perfect API. It could just get two props like this for example `EditCardForm({onSubmit, initialValues})`
The new prop could be used like this:
`defaultValue={initialValues?.[attribute] ? initialValues[attribute] : null}`

Reply:
The two props form and activeCar have these disadvantages:
1. They are highly dependent on each other and the component crashes if an outside component hands in non matching props (for example form="edit" without handing over an activeCar prop)
2. the activeCar prop gives the component information about the outside world and how it is being used by the props name

Reply:
Done, thanks for the suggestions